test/integration/style-spec/tests/sources.input.json

```diff
@@ -64,6 +64,42 @@
       "tiles": ["http://example.com/{x}/{y}/{z}.png"],
       "valueOf": null,
       "toLocaleString": 6
+    },
+    "source-vector-default": {
+      "type": "vector",
+      "url": "https://demotiles.maplibre.org/tiles/tiles.json"
+    },
+    "source-vector-mvt": {
+      "type": "vector",
+      "url": "https://demotiles.maplibre.org/tiles/tiles.json",
+      "encoding": "mvt"
+    },
+    "source-vector-mlt": {
+      "type": "vector",
+      "url": "https://demotiles.maplibre.org/tiles/tiles.json",
+      "encoding": "mlt"
+    },
+    "source-vector-custom": {
+      "type": "vector",
+      "url": "https://demotiles.maplibre.org/tiles/tiles.json",
+      "encoding": "custom"
+    },
+    "source-vector-blank": {
+      "type": "vector",
+      "url": "https://demotiles.maplibre.org/tiles/tiles.json",
+      "encoding": ""
+    },
+    "source-vector-numeric": {
+      "type": "vector",
+      "url": "https://demotiles.maplibre.org/tiles/tiles.json",
+      "encoding": 3
+    },
+    "source-vector-array": {
+      "type": "vector",
+      "url": "https://demotiles.maplibre.org/tiles/tiles.json",
+      "encoding": [
+        "mvt"
+      ]
```

Comment:
The tests only check tilejson usage. Can you add usage with a direct xyz tile URL?